test_common.go

```diff
@@ -44,3 +118,91 @@ func mustHexDecode(t *testing.T, s string) []byte {
 	}
 	return data
 }
+
+func signerFromJwKey(t *testing.T, jkey jwk.Key) cose.Signer {
+	var (
+		key crypto.Signer
+		crv elliptic.Curve
+		alg cose.Algorithm
+	)
+	err := jkey.Raw(&key)
+	require.NoError(t, err)
+	switch v := key.(type) {
+	case *ecdsa.PrivateKey:
+		crv = v.Curve
+		if crv == elliptic.P256() {
+			alg = cose.AlgorithmES256
+			break
+		} else if crv == elliptic.P384() {
+			alg = cose.AlgorithmES384
+			break
+		}
+		require.True(t, false, "unknown elliptic curve %v", crv)
+	default:
+		require.True(t, false, "unknown private key type %v", reflect.TypeOf(key))
+	}
+	s, err := cose.NewSigner(alg, key)
+	require.Nil(t, err)
+
+	return s
+}
+
+func pubKeyFromJwKey(t *testing.T, jkey jwk.Key) crypto.PublicKey {
+	var key crypto.Signer
+	err := jkey.Raw(&key)
+	require.NoError(t, err)
+	vk := key.Public()
+	return vk
+}
+
+func signerFromJWK(t *testing.T, j string) cose.Signer {
+	alg, key := getAlgAndKeyFromJWK(t, j)
+	s, err := cose.NewSigner(alg, key)
+	require.Nil(t, err)
+
+	return s
+}
+
+func getAlgAndKeyFromJWK(t *testing.T, j string) (cose.Algorithm, crypto.Signer) {
+	ks, err := jwk.ParseString(j)
+	require.Nil(t, err)
+
+	var (
+		key crypto.Signer
+		crv elliptic.Curve
+		alg cose.Algorithm
+	)
+
+	k, ok := ks.Get(0)
+	require.True(t, ok)
+	err = k.Raw(&key)
+	require.NoError(t, err)
+
+	switch v := key.(type) {
+	case *ecdsa.PrivateKey:
+		crv = v.Curve
+		if crv == elliptic.P256() {
+			alg = cose.AlgorithmES256
+			break
+		}
+		require.True(t, false, "unknown elliptic curve %v", crv)
+	default:
+		require.True(t, false, "unknown private key type %v", reflect.TypeOf(key))
+	}
+	return alg, key
+}
+
+func pubKeyFromJWK(t *testing.T, j string) crypto.PublicKey {
+	_, key := getAlgAndKeyFromJWK(t, j)
+	vk := key.Public()
+	return vk
+}
+
+func getJwkKeyFromPemKey(testKey []byte) (jwk.Key, error) {
+	key, err := jwk.ParseKey(testKey, jwk.WithPEM(true))
+	if err != nil {
+		return nil, err
+	}
+	fmt.Printf("received key = %s", key)
```

Comment:
this can be dropped (besides, I don't think the `%s` formatter would make sense of the jwk object)